src/sailfishui/plugin/pendinguploadsmodel.cpp

```diff
@@ -129,7 +129,13 @@ void PendingUploadsModel::setData(const QStringList &data)
         item.filePath = filePath;
         item.dateCreated = QFileInfo(filePath).created();
 
-        d->contents.append(item);
+        int i = 0;
+        for (; i < d->contents.count(); ++i) {
+            if (item.dateCreated > d->contents.at(i).dateCreated) {
+                break;
+            }
+        }
+        d->contents.insert(i, item);
```

Comment:
Now the beginInsertRows values would be wrong if the insertions don't happen at the end?

Reply:
Oops, fixed now.